parallel-consumer-core/src/main/java/io/confluent/parallelconsumer/offsets/OffsetEncoding.java

```diff
@@ -39,7 +39,15 @@ public enum OffsetEncoding {
      * switch from encoding run lengths as Shorts to Integers
      */
     RunLengthV2(v2, (byte) 'e'),
-    RunLengthV2Compressed(v2, (byte) 'p');
+    RunLengthV2Compressed(v2, (byte) 'p'),
+
+    /**
+     * checks for pre-existing Kafka Streams metadata
+     * https://github.com/apache/kafka/blob/cc77a38d280657a0e3969b255f103af4d11c7914/streams/src/main/java/org/apache/kafka/streams/processor/internals/TopicPartitionMetadata.java#L33
```

Comment:
good hunting!
```suggestion
     * <a href="https://github.com/apache/kafka/blob/cc77a38d280657a0e3969b255f103af4d11c7914/streams/src/main/java/org/apache/kafka/streams/processor/internals/TopicPartitionMetadata.java#L33">source from Kafka Streams code</a>
```

or something like that

Reply:
actually as this is repeated, I would actually prefer to just refer to the documentation in the dedicated exception in KafkaStreamsEncodingNotSupported

```suggestion
     * @see KafkaStreamsEncodingNotSupported
     * ```